g_checker_for_itf/main.py

```diff
@@ -0,0 +1,196 @@
+        prog="gchk",
+        description="A checker if your credits meet the graduation requirements or not.",
+        formatter_class=(
+            lambda prog: GchkHelpFormatter(
+                prog,
+                **{
+                    "width": shutil.get_terminal_size(fallback=(120, 50)).columns,
+                    "max_help_position": 30,
+                },
+            )
+        ),
+    )
+    parser.add_argument(
+        "-i",
+        "--input",
+        help="target file from twins (UTF-8, CSV)",
+        default="sample.csv",
+    )
+    parser.add_argument(
+        "-r", "--requirements", help="requirements file", default="coins20.json"
+    )
+    parser.add_argument("-g", "--gpa", help="print GPA", action="store_true")
+    parser.add_argument("-d", "--drop", help="print drop credit", action="store_false")
+    parser.add_argument("-n", "--name", help="print name and id", action="store_true")
+    parser.add_argument("-s", "--save", help="save as JSON", action="store_true")
+    parser.add_argument("-e", "--expect", help="count 履修中", action="store_true")
+    parser.add_argument(
+        "-V",
+        "--version",
+        action="version",
+        version=f"%(prog)s {__version__}",
+    )
+    return parser.parse_args()
+
+
+def parseJSON(JSONFILENAME: str) -> Level1:
+    data = json.load(open(JSONFILENAME, "r"))
+    MAX_STR = "max_certificated_credit_num"
+    MIN_STR = "min_certificated_credit_num"
+
+    if len(data) == 0:
+        raise ValueError("json data is empty")
+    else:
+        first_key, *_ = data
+        init_k1, init_v1 = first_key, data[first_key]
+        all = Level1(init_k1, init_v1[MAX_STR], init_v1[MIN_STR])
+
+    for k1, v1 in data.items():
+        all = Level1(k1, v1[MAX_STR], v1[MIN_STR])
+        for k2, v2 in v1["leaf"].items():
+            all.add(Level2(k2, v2[MAX_STR], v2[MIN_STR]))
+            for k3, v3 in v2["leaf"].items():
+                cf_k2 = cast(Dir, all.course_filter[k2])
+                cf_k2.add(Level3(k3, v3[MAX_STR], v3[MIN_STR]))
+                for k4, v4 in v3["leaf"].items():
+                    cf_k3 = cast(Dir, cf_k2.course_filter[k3])
+                    cf_k3.add(Level4(k4, v4[MAX_STR], v4[MIN_STR]))
+                    for k5, v5 in v4["leaf"].items():
+                        cf_k4 = cast(Dir, cf_k3.course_filter[k4])
+                        cf_k4.add(Level5(k5, v5[MAX_STR], v5[MIN_STR]))
+                        cf_k5 = cast(Dir, cf_k4.course_filter[k5])
+                        cf_k5.add(
+                            RecognizedFilter(
+                                k5,
+                                v5["leaf"]["regexp_number"],
+                                v5["leaf"]["regexp_name"],
+                            )
+                        )
+
+    return all
+
+
+def gp(ls: List[KamokuClass]) -> None:
+    gps = 0.0
+    credit_sum = 0.0
+    grade_points = {"A+": 4.3, "A": 4, "B": 3, "C": 2, "D": 1}
```

Comment:
```suggestion
    grade_points = {"A+": 4.3, "A": 4, "B": 3, "C": 2, "D": 0}
```
D についてのグレードは 0 のようです。
https://www.tsukuba.ac.jp/education/ug-courses-gpa/

Reply:
https://github.com/oshamashama/g-checker-for-itf/blob/b6ac72838e50763dc2800eddf366f0c725dd6115/main.py#L243-L258

ここでは